src/mudlet.cpp

```diff
@@ -1575,32 +1575,40 @@ void mudlet::slot_timer_fires()
     if (!pQT) {
         return;
     }
-    if (mTimerMap.contains(pQT)) {
-        TTimer* pTT = mTimerMap[pQT];
+
+    // Pull the Host name and TTimer::id from the properties:
+    QString hostName(pQT->property(TTimer::scmProperty_HostName).toString());
+    if (hostName.isEmpty()) {
+        qWarning().nospace().noquote() << "mudlet::slot_timer_fires() INFO - Host name is empty - so TTimer has probably been deleted.";
+        pQT->deleteLater();
+        return;
+    }
+
+    Host* pHost = mHostManager.getHost(hostName);
+    Q_ASSERT_X(pHost, "mudlet::slot_timer_fires()", "Unable to deduce Host pointer from data in QTimer");
+    int id = pQT->property(TTimer::scmProperty_TTimerId).toInt();
+    if (!id) {
+        qWarning().nospace().noquote() << "mudlet::slot_timer_fires() INFO - TTimer ID is zero - so TTimer has probably been deleted.";
+        pQT->deleteLater();
+        return;
+    }
+    TTimer* pTT = pHost->getTimerUnit()->getTimer(id);
+    if (pTT) {
+        qDebug().nospace().noquote() << "mudlet::slot_timer_fires() INFO - Host: \"" << hostName << "\" QTimer firing for TTimer Id:" << id;
+        qDebug().nospace().noquote() << "    (objectName:\"" << pQT->objectName() << "\")";
```

Comment:
We probably should remove this debug output, it will get super spammy. My system for example has tons of timers running - and someone with an animated framework will have timers running every 16ms!

Reply:
I'll comment it out for now - I am aware of another timer issue (every-time a timer is selected and viewed in the editor it gets saved **even if no changes are made** - and the saving process *restarts any active timers* which can be detrimental to user code that is using timers {particularly for long durations} for anything critical) and I may need it back again...

Reply:
Done - just waiting on the CI before I ***Squash and merge*** this.